src/vs/vscode.proposed.d.ts

```diff
@@ -2276,20 +2276,58 @@ declare module 'vscode' {
 
 	//#region https://github.com/Microsoft/vscode/issues/15178
 
-	// TODO@API must be a class
-	// TODO@API @lramos15 Call this XYZTabs
-	export interface OpenEditorInfo {
-		//viewColumn: ViewColumn; // todo @lramos15
-		name: string;
-		resource: Uri; // make optional @lramos15
+	/**
+	 * Represents a tab within the window
+	 */
+	export interface Tab {
+		/**
+		 * The text displayed on the tab
+		 */
+		label: string;
+
+		/**
+		 * The position of the tab
+		 */
+		viewColumn: ViewColumn;
+
+		/**
+		 * The resource represented by the tab.
+		 * May be undefined in certain cases such as diff editor and marketplace tabs
+		 */
+		resource?: Uri;
+
+		/**
+		 * Whether or not the tab is currently active
+		 * Dictated by being the selected tab in the active group
+		 */
 		isActive: boolean;
 	}
 
 	export namespace window {
-		export const openEditors: ReadonlyArray<OpenEditorInfo>;
+		/**
+		 * A list of all opened tabs
+		 * Ordered from left to right
+		 */
+		export const tabs: ReadonlyArray<Tab>;
+
+		/**
+		 * The currently active tab
+		 * Undefined if no tabs are currently opened
+		 */
+		export const activeTab: Tab | undefined;
+
+		/**
+		 * An {@link Event} which fires when the array of {@link window.tabs tabs}
+		 * has changed.
+		 */
+		export const onDidChangeTabs: Event<ReadonlyArray<Tab>>;
```

Comment:
`readonly Tab[]` and consider an event which contains added, changed, removed tabs, similar to [WorkspaceFoldersChangeEvent](https://github.com/microsoft/vscode/blob/8011d05df67c93228b4c0e1ded606cb98be4de09/src/vs/vscode.d.ts#L10437-L10437)

Reply:
I wonder the usefulness of that. I was just going off the `onDidChangeVisibleTextEditors` which returns the text editors array.

Reply:
It will likely help you manage diagnostics, eg. remove diagnostics for closed tab. Have this information in the event means user don't need to iterate over the array all the time. Tho, this is P2 and not blocking merging this

Reply:
I feel like if we were to add this we would definitely want to add an `order` as then it would be a lot harder to discern where these tabs were located

Reply:
You need to explain why that motivates `order` and what order-value closed tabs should have?

Reply:
For example a rename is basically just an open and close of a tab in the same location. It also could be interesting to know what tab now possesses the location of the previously closed tab. The order value would just be the index it had in the group. So it would be something like `[{order: 0, viewColumn: 1}, {order: 1, viewColumn: 1}, {order: 0, viewColumn: 2}, {order: 1, viewColumn: 2}]`

Additionally I either think we would want to have Added, Changed, Removed on the onDidChange or break up the events. onDidChange would just be the list, then onDidAdd, onDidRemove, onDidMove, onDidRename etc because there's a lot more than adding and removing a tab

Reply:
So, you do agree that having knowledge about closed tabs is useful (sample [here](https://github.com/microsoft/vscode/pull/131998#discussion_r700264897)) but you couple its value to an order attribute? I think that is just artificial complexity and I wanna see a real world use-case

Reply:
I'm not sure of a use case off the top of my head by I know added / removed is a common API pattern and I value the consistency. I will bring it up as a discussion for the API sync and we can discuss there. I'm going to merge this since this is first major chunk of the API and another PR will be used for adjustments